scripts/tasks/preprocess-api-history.ts

```diff
@@ -0,0 +1,232 @@
+import globby from 'globby';
+import path from 'path';
+import { fromMarkdown } from 'mdast-util-from-markdown';
+import { visit } from 'unist-util-visit';
+import type { Html, Root } from 'mdast';
+import type { Node, Literal } from 'unist';
+import Ajv, { JSONSchemaType, ValidateFunction } from 'ajv';
+import { readFile, writeFile } from 'fs/promises';
+import { fromHtml } from 'hast-util-from-html';
+import logger from '@docusaurus/logger';
+import { parse as parseYaml } from 'yaml';
+import { isHtml } from '@site/src/util/mdx-utils';
+
+const apiHistoryRegex = /<!--[\s\S]*?(```[\s\S]*?```)[\s\S]*?-->/;
+
+interface ChangeSchema {
+  'pr-url': string;
+  'breaking-changes-header'?: string;
+  description?: string;
+}
+
+interface ApiHistory {
+  added?: ChangeSchema[];
+  deprecated?: ChangeSchema[];
+  changes?: ChangeSchema[];
+}
+
+interface LiteralString extends Literal {
+  value: string;
+}
+
+// Copied from here: <https://github.com/electron/website/blob/feat/api-history/scripts/tasks/add-frontmatter.ts#L16-L23>
+const getMarkdownFiles = async (startPath: string) => {
+  const filesPaths = await globby(path.posix.join(startPath, 'api', '/*.md'));
+
+  const files: Map<string, string> = new Map();
+  for (const filePath of filesPaths) {
+    const content = await readFile(filePath, 'utf-8');
+    files.set(filePath, content);
+  }
+
+  return files;
+};
+
+// Copied from here but slightly different: <https://github.com/electron/lint-roller/blob/bac245478ba69017c2a82e8ba1b2884a37647494/bin/lint-markdown-api-history.ts#L42-L67>
+function findPossibleApiHistoryBlocks(tree: Root) {
+  const codeBlocks: Html[] = [];
+
+  visit(
+    tree,
+    (node: Node) =>
+      isHtml(node) &&
+      (node as LiteralString).value.includes('```') &&
+      (node as LiteralString).value.toLowerCase().includes('yaml') &&
+      (node as LiteralString).value.toLowerCase().includes('history'),
+    (node) => {
+      if (!isHtml(node)) return; // Should never happen, but TypeScript needs this check
+      codeBlocks.push(node);
```

Comment:
```suggestion
      codeBlocks.push(node as Html);
```

Reply:
Decided to revert in Slack meeting, [`6e74b23` (#594)](https://github.com/electron/website/pull/594/commits/6e74b234bd6c901a78cc54870e56639db9dc0ed9)